File: apps/api/schemas/analysis.py
from pydantic import BaseModel, Field


class ContentStructurePart(BaseModel):
    part: str
    description: str


class VideoStyleAnalysisResponse(BaseModel):
    videoId: str
    hookType: str | None = None
    hookText: str | None = None
    topicCategory: str | None = None
    targetAudience: list[str] = Field(default_factory=list)
    contentStructure: list[ContentStructurePart] = Field(default_factory=list)
    emotionalTone: str | None = None
    commonPhrases: list[str] = Field(default_factory=list)
    endingType: str | None = None
    shootingStyle: str | None = None
    reusableTemplate: str | None = None


class TranscriptSummary(BaseModel):
    fullText: str
    language: str


class VideoAnalysisResponse(BaseModel):
    videoId: str
    transcript: TranscriptSummary | None = None
    analysis: VideoStyleAnalysisResponse | None = None

File: packages/shared-types/src/analysis.ts
export interface TranscriptSegment {
  start: number;
  end: number;
  text: string;
}

export interface Transcript {
  videoId: string;
  language: string;
  fullText: string;
  segments: TranscriptSegment[];
  words: unknown[];
  asrModel: string;
}

export interface ContentStructurePart {
  part: string;
  description: string;
}

export interface VideoStyleAnalysis {
  videoId?: string;
  hookType: string;
  hookText: string;
  topicCategory: string;
  targetAudience: string[];
  contentStructure: ContentStructurePart[];
  emotionalTone: string;
  commonPhrases: string[];
  endingType: string;
  shootingStyle: string;
  reusableTemplate: string;
  subtitlePosition?: string;
  subtitleStyle?: string;
  subtitleConsistency?: string;
}

export interface FrameAnalysis {
  frameTime: number;
  shotType?: string;
  cameraAngle?: string;
  composition?: string;
  background?: string;
  subtitleVisible?: boolean;
  subtitlePosition?: string;
  subtitleStyle?: string;
  visualElements?: string[];
  bRoll?: boolean;
}

export interface VisualAnalysis {
  videoId?: string;
  frames: FrameAnalysis[];
  summary: Record<string, unknown>;
  visionModel: string;
}
